### PR TITLE
docs: add cli documentation

### PR DESCRIPTION
**Pull request by osergioneto**
Addition of section sdkgen/cli.
- CLI: command line interface;
- Example;
- Options;
- Targets;

### CODE REVIEW

File: cli.mdx [hunk 1]
```diff
@@ -0,0 +1,39 @@
+---
+name: CLI   
+route: /cli
+---
+
+# CLI - Command Line Interface
+
+Tool to help generate files from sdkgen to specific targets.
+
```
Comment (dgadelha) [inline cli.mdx]: You should mention the installation prerequisites and the how to:

- A current/LTS version of Node.js is required;
- It should be installed with `npm i -g @sdkgen/cli`

I'd also attempt to rewrite that line to add more information about what the CLI does.

File: cli.mdx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+## Targets
+
+| param | type |
+| :-- | :-: |
+| --typescript_nodeserver | Server Typescript Node 
+| --typescript_nodeclient | Client Typescript Node 
+| --typescript_web | Typescript Web 
+| --typescript_interfaces | Interfaces Typescript
+| --flutter | Flutter
+| --csharp_server | Server C#
```
Comment (dgadelha) [inline cli.mdx]: What about describing a bit more about the runtime?

```suggestion
| --typescript_nodeserver | TypeScript Server for Node.js runtime
| --typescript_nodeclient | TypeScript Client for Node.js runtime
| --typescript_web | TypeScript Web Client for Browser runtime
| --typescript_interfaces | TypeScript Interfaces for API Types
| --flutter | Flutter Client for Dart runtime
| --csharp_server | C# Server for .NET Core runtime
```

Comment (dgadelha) [reply cli.mdx]: Been thinking in creating a table with the columns:

- param
- type (Client / Server)
- language
- runtime

or splitting into 2 tables (client and server)

**Review by dgadelha — CHANGES_REQUESTED** [review]

File: cli.mdx (hunk 2 above)
Comment (lbguilherme) [reply cli.mdx]: Also `--typescript_nodeserver` is incorrect. It should be `-t typescript_nodeserver`

File: doczrc.js [hunk 3]
```diff
@@ -4,6 +4,7 @@ export default {
         "Intro",
         "Getting Started",
         "Defining your schema",
-        "Encode and Decode"
+        "Encode and Decode",
+        "CLI"
```
Comment (lbguilherme) [inline doczrc.js]: Move it before "Defining your schema"

File: cli.mdx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+```
+
+## Options
+
+| param | type |  |
+| :-- | :-: | :-: |
+| --source | string | Specifies the source file
+| -o, --output | string | Specifies the output file 
+| -t, --target | string | Specifies the target platform and language 
+| -h, --help | string | Display usage guide
```
Comment (dgadelha) [inline cli.mdx:32]: Misses [compatibility check command](https://github.com/sdkgen/sdkgen/blob/master/cli/src/compatibility.ts)

File: cli.mdx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+## Targets
+
+| param | type | language | runtime
+| :-- | :-- | :-- | :--
+| typescript_nodeserver | Server | TypeScript | Node.js
+| typescript_nodeclient | Client | TypeScript | Node.js
+| typescript_web | Client | TypeScript | Web Browser 
+| flutter | Client | Dart | Flutter
+| csharp_server | Server | C# | .NET Core
```
Comment (dgadelha) [inline cli.mdx:42]: Missing:

- kotlin_android
- typescript_interfaces

**Review by dgadelha — CHANGES_REQUESTED** [review]

**Pull request closed without merging**